espnet2/asr/espnet_model.py

```diff
@@ -81,32 +84,60 @@ def __init__(
         self.preencoder = preencoder
         self.postencoder = postencoder
         self.encoder = encoder
-        # we set self.decoder = None in the CTC mode since
-        # self.decoder parameters were never used and PyTorch complained
-        # and threw an Exception in the multi-GPU experiment.
-        # thanks Jeff Farris for pointing out the issue.
-        if ctc_weight == 1.0:
-            self.decoder = None
-        else:
+
+        self.use_transducer_decoder = joint_network is not None
+
+        self.error_calculator = None
+
+        if self.use_transducer_decoder:
```

Comment:
I thought we would still have the option to apply transducer decoder and attention decoder at the same time, maybe we can just allow them to stand together instead of only allowing one at a time?

Reply:
I'm not sure I understand why we would want both decoder at the same time outside multi-decoding or some rescoring? Do you have an idea?
Previous implementation didn't consider this case, the unification was done to allow RNN-T decoder w/ attention. However, the performance gain is minimal and it would severely impact decoding speed.

Reply:
Okay, I was at the beginning thought you would want to conduct some experiments with the setting of multi-decoder given the unification. Seems that I just misunderstood that. Then, everything looks cool! thanks